Updated Documentation with examples using props (#12838)
remix-run/react-router: docs/start/framework/route-module.md

  );
}
```
### Default props passed to the Component

> 1. params: An object containing the route parameters (if any).

Reviewer: ```suggestion
1. `params`: An object containing the route parameters (if any).
```

can wrap these in code formatting

2. location: An object providing details about the current location (e.g., pathname, search, etc.).

3. navigate: A function to programmatically navigate to another route.

4. loaderData: The data returned from the loader function assigned to the route. eg. A loader might fetch some data from an API and return it for the route to use.

### Using props
```tsx filename=app/routes/my-route-with-default-params.tsx
export default function MyRouteComponent({ loaderData, actionData, params, matches }) {
  return (
    <div>
      <h1>Welcome to My Route with Props!</h1>
      <p>Loader Data: {JSON.stringify(loaderData)}</p>
      <p>Action Data: {JSON.stringify(actionData)}</p>
      <p>Route Parameters: {JSON.stringify(params)}</p>
      <p>Matched Routes: {JSON.stringify(matches)}</p>
    </div>
  );
}
```

### Benefits of using props

> 1. Route data is directly accessible via props, leading to more readable code.

Reviewer: Another benefit is Route data can be automatically typed if you follow the instructions and import the `Route` from the `+types`

> 1. Route data is directly accessible via props, leading to more readable code.

Reviewer: Another benefit is Route data can be automatically typed if you follow the instructions and import the Route from the +types

  );
}
```
### Default props passed to the Component
1. `params`: An object containing the route parameters (if any).

> 2. location: An object providing details about the current location (e.g., pathname, search, etc.).

Reviewer: ```suggestion
2. `location`: An object providing details about the current location (e.g., pathname, search, etc.).
```

etc

3. navigate: A function to programmatically navigate to another route.

4. loaderData: The data returned from the loader function assigned to the route. eg. A loader might fetch some data from an API and return it for the route to use.

### Using props
```tsx filename=app/routes/my-route-with-default-params.tsx

> export default function MyRouteComponent({ loaderData, actionData, params, matches }) {

Reviewer: ```suggestion
import type { Route } from './+types/route-name'

export default function MyRouteComponent({ loaderData, actionData, params, matches }) {
```

  );
}
```

> ### Default props passed to the Component
> 1. `params`: An object containing the route parameters (if any).
> 
> 2. `location`: An object providing details about the current location (e.g., pathname, search, etc.).
> 
> 3. `navigate`: A function to programmatically navigate to another route.
> 
> 4. `loaderData`: The data returned from the loader function assigned to the route. eg. A loader might fetch some data from an API and return it for the route to use.
> 
> ### Using props
> ```tsx filename=app/routes/my-route-with-default-params.tsx
> import type { Route } from './+types/route-name'
> 
> export default function MyRouteComponent({ 
>   loaderData, 
>   actionData,
>   params, 
>   matches 
> }: Route.ComponentProps) {
>   return (
>     <div>
>       <h1>Welcome to My Route with Props!</h1>
>       <p>Loader Data: {JSON.stringify(loaderData)}</p>
>       <p>Action Data: {JSON.stringify(actionData)}</p>
>       <p>Route Parameters: {JSON.stringify(params)}</p>
>       <p>Matched Routes: {JSON.stringify(matches)}</p>
>     </div>
>   );
> }
> ```
> 
> ### Benefits of using props
> 1. Route data is directly accessible via props, leading to more readable code.
> 2. Route data can be automatically typed when Route is imported from `+types`.

Reviewer 2: ```suggestion
### Props passed to the Component

When the component is rendered, it is provided the props defined in `Route.ComponentProps` that React Router will automatically generate for you. These props include: 

1. `loaderData`: The data returned from the `loader` function in this route module
2. `actionData`: The data returned from the `action` function in this route module
3. `params`: An object containing the route parameters (if any).
4. `matches`: An array of all the matches in the current route tree.

You can use these props in place of hooks like `useLoaderData` or `useParams`. This may be preferrable because they will be automatically typed correctly for the route. 

### Using props
```tsx filename=app/routes/my-route-with-default-params.tsx
import type { Route } from './+types/route-name'

export default function MyRouteComponent({ 
  loaderData, 
  actionData,
  params, 
  matches 
}: Route.ComponentProps) {
  return (
    <div>
      <h1>Welcome to My Route with Props!</h1>
      <p>Loader Data: {JSON.stringify(loaderData)}</p>
      <p>Action Data: {JSON.stringify(actionData)}</p>
      <p>Route Parameters: {JSON.stringify(params)}</p>
      <p>Matched Routes: {JSON.stringify(matches)}</p>
    </div>
  );
}
```
```

Fixed up the actual props that are passed (`location` and `navigate` aren't in there) and clarified some of the language.
